### PR TITLE
Simplify system tests

### PR DESCRIPTION
**Pull request by DominicKramer**
Fixes: #462

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
Perhaps a silly question, but why not use async/await?

**Comment by DominicKramer** [comment]
@JustinBeckwith Sorry I'm still in the process of updating this PR.  I wanted to first have the tests pass with the current layout and then update the PR to use async/await.

**Comment by DominicKramer** [comment]
PTAL

File: system-test/test-e2e.ts [hunk 1]
```diff
@@ -141,307 +136,218 @@ describe('@google-cloud/debug end-to-end behavior', () => {
...
+
+    const breakpoints = await api.listBreakpoints(debuggeeId!, {});
+    console.log('-- List of breakpoints\n', breakpoints);
+
+    // Delete every breakpoint
+    const promises = breakpoints.map((breakpoint: stackdriver.Breakpoint) => {
+      return api.deleteBreakpoint(debuggeeId!, breakpoint.id);
+    });
+
+    await Promise.all(promises);
```
Comment (ofrobots) [inline system-test/test-e2e.ts]: Reuse `verifyDeleteBreakpoints` here?

File: system-test/test-e2e.ts [hunk 2]
```diff
@@ -141,307 +136,218 @@ describe('@google-cloud/debug end-to-end behavior', () => {
...
+    // Check that the debuggee created in this test is among the list of
+    // debuggees, then list its breakpoints
+    console.log(
+        '-- List of debuggees\n', util.inspect(debuggees, {depth: null}));
+    assert.ok(debuggees, 'should get a valid ListDebuggees response');
+
+    const result = _.find(debuggees, (d: Debuggee) => {
+      return d.id === debuggeeId;
+    });
+    assert.ok(result, 'should find the debuggee we just registered');
```
Comment (ofrobots) [inline system-test/test-e2e.ts]: Reuse `verifyDebuggeeFound` here?

File: system-test/test-e2e.ts [hunk 3]
```diff
@@ -141,307 +136,218 @@ describe('@google-cloud/debug end-to-end behavior', () => {
...
+      id: 'breakpoint-3',
+      location: {path: FILENAME, line: 5},
+      condition: 'n === 10',
+      action: 'LOG',
+      expressions: ['o'],
+      logMessageFormat: 'o is: $0',
+      stackFrames: [],
+      evaluatedExpressions: [],
+      variableTable: []
+    });
```
Comment (ofrobots) [inline system-test/test-e2e.ts]: Rename `verifySetLogpoint` to `verifySetBreakpoint` and make it accept a parameter – the breakpoint. Reuse the function to replace the code here.

File: system-test/test-e2e.ts [hunk 4]
```diff
@@ -141,307 +136,218 @@ describe('@google-cloud/debug end-to-end behavior', () => {
...
+          (child.transcript.match(/LOGPOINT: o is: \{"a":\[1,"hi",true\]\}/g) ||
+           []).length;
+      // A log count of greater than 10 indicates that we did not
+      // successfully pause when the rate of `maxLogsPerSecond` was
+      // reached.
+      assert(logCount <= 10, 'log count is greater than 10: ' + logCount);
+      // A log count of less than 3 indicates that we did not successfully
+      // resume logging after `logDelaySeconds` have passed.
+      assert(logCount > 2, 'log count is not greater than 2: ' + logCount);
+    });
```
Comment (ofrobots) [inline system-test/test-e2e.ts]: Similarly reuse `verifyHitBreakpoint` here if possible.

File: system-test/test-e2e.ts (hunk 2 above)
Comment (DominicKramer) [reply system-test/test-e2e.ts]: Done

File: system-test/test-e2e.ts (hunk 1 above)
Comment (DominicKramer) [reply system-test/test-e2e.ts]: Done

File: system-test/test-e2e.ts (hunk 3 above)
Comment (DominicKramer) [reply system-test/test-e2e.ts]: Done

File: system-test/test-e2e.ts (hunk 4 above)
Comment (DominicKramer) [reply system-test/test-e2e.ts]: I simplified the code the most I think I should.

**Pull request merged**